ipa-core/src/protocol/dp/distributions.rs

```diff
@@ -63,44 +58,92 @@ impl Distribution<f64> for RoundedBoxMuller {
         self.inner.sample(rng).round()
     }
 }
-
 impl From<BoxMuller> for RoundedBoxMuller {
     fn from(value: BoxMuller) -> Self {
         Self { inner: value }
     }
 }
-
+/// Double Geometric Distribution
+///
+/// Generates a sample from a geometric distribution with the given success probability.
+fn generate_geometric<R: Rng + ?Sized>(probability: f64, rng: &mut R) -> isize {
+    // Create a Bernoulli distribution with the specified success probability
+    let bernoulli = Bernoulli::new(probability).expect("Invalid probability");
+    // Generate Bernoulli random numbers until the first success
+    // let mut rng = rand::thread_rng();
+    let mut attempts = 0;
+    // let sample = ud.sample(rng);
+    while !bernoulli.sample(rng) {
+        attempts += 1;
+    }
+    attempts
```

Comment:
this can't be negative, so maybe a fixed size integer is more appropriate here

Reply:
changed to u32